payload/Library/Scripts/Install or Defer.sh

```diff
@@ -621,14 +624,16 @@ MSG_INSTALL="$(echo "$MSG_INSTALL" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT
 MSG_INSTALL_NOW="$(echo "$MSG_INSTALL_NOW" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT_CONTACT}/")"
 MSG_UPDATING="$(echo "$MSG_UPDATING" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT_CONTACT}/")"
 
-# Check for updates, exit if none found, otherwise continue.
-check_for_updates
-
 # Perform first-run tasks, including calculating deadline.
 FORCE_DATE=$(/usr/bin/defaults read "$PLIST" UpdatesForcedAfter 2>"/dev/null")
 if [[ -z "$FORCE_DATE" || "$FORCE_DATE" -gt $(( $(/bin/date +%s) + MAX_DEFERRAL_TIME )) ]]; then
+    # Check for updates, exit if none found, otherwise continue.
+    check_for_updates
     FORCE_DATE=$(( $(/bin/date +%s) + MAX_DEFERRAL_TIME ))
     /usr/bin/defaults write "$PLIST" UpdatesForcedAfter -int "$FORCE_DATE"
+else 
+    # Read in the update_list from the preferences file
+    UPDATE_LIST=$(/usr/bin/defaults read "$PLIST" UPDATE_LIST 2>"/dev/null")
```

Comment:
My issue with this change is that there's no validation occurring of whether the pending updates are still needed between script runs. As your script update stands, if I defer the update prompt, then run updates on my own between prompts, I'll keep getting prompted until the deadline is reached, even if there are no more updates to run, since the script keeps reading now-outdated information from a static plist entry. I recognize that the goal is to reduce `softwareupdate --list` runs, but I worry that this will create a different sort of negative user experience of false-positive alerting.